terraform/dev/main.tf

```diff
@@ -0,0 +1,107 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+##
+# Data providers
+##
+data "aws_vpc" "selected" {
+  id = "${var.vpc_id}"
+}
+
+data "aws_subnet_ids" "app" {
+  vpc_id = "${var.vpc_id}"
+
+  tags {
+    Layer       = "app"
+    stack       = "${var.stack}"
+    application = "${var.app}"
+  }
+}
+
+data "aws_subnet" "app" {
+  count = "${length(data.aws_subnet_ids.app.ids)}"
+  id    = "${data.aws_subnet_ids.app.ids[count.index]}"
+}
+
+data "aws_elb" "elb" {
+  name = "${var.elb_name}"
+}
+
+##
+# Security groups
+##
+resource "aws_security_group" "ci" {
+  name        = "ci-to-app-servers"
+  description = "Allow CI access to app servers"
+  vpc_id      = "${data.aws_vpc.selected.id}"
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["${var.ci_cidrs}"]
+  }
+}
+
+##
+# Launch configuration
+##
+resource "aws_launch_configuration" "app" {
+  security_groups = [
+    "${var.app_sg_id}",
+    "${var.vpn_sg_id}",
+    "${aws_security_group.ci.id}",
+  ]
+
+  key_name                    = "${var.key_name}"
+  image_id                    = "${var.ami_id}"
+  instance_type               = "${var.instance_type}"
+  associate_public_ip_address = false
+  name_prefix                 = "bb-${var.stack}-app-"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+##
+# Autoscaling group
+##
+resource "aws_autoscaling_group" "main" {
```

Comment:
I know I already mentioned this in chat, but I saw you already have an auto scaling group configuration for a different project, and this may be a chance to reuse it (it has Cloudwatch alarms built in, and probably works with AdHoc's other deployment tools).  That's not open source at the moment, but there's nothing secret about it (even copying it in here might be worth it in the short term).

Reply:
I will look into this!